Source/ApplicationServices/LibraryCommands.cs

```diff
@@ -27,10 +27,17 @@ static LibraryCommands()
 			ScanEnd += (_, __) => Scanning = false;
 		}
 
-        public static async Task<List<LibraryBook>> FindInactiveBooks(Func<Account, Task<ApiExtended>> apiExtendedfunc, List<LibraryBook> existingLibrary, params Account[] accounts)
+        public static async Task<List<LibraryBook>> FindInactiveBooks(Func<Account, Task<ApiExtended>> apiExtendedfunc, IEnumerable<LibraryBook> existingLibrary, params Account[] accounts)
 		{
 			logRestart();
 
+			lock (_lock)
+			{
+				if (Scanning)
+					return new();
+				ScanBegin?.Invoke(null, accounts.Length);
```

Comment:
Not sure about this one. I copied your code ImportAccountAsync, but I remember your warning about not doing this on my last PR.  You good with these?

Reply:
I'd just put the event invocation after the lock. Did I screw this up somewhere?

Reply:
Yeah, one method down. I just copied you lol

Reply:
Darn. I don't want to step on your current work